File: Data Structures & Algorithms/insert-new-interval/submission-5.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> solution;
        int i = 0;

        while ((i < intervals.size()) && (intervals[i][1] < newInterval[0])) {
            solution.push_back(intervals[i++]);
        }

        while ((i < intervals.size()) && (newInterval[1] >= intervals[i][0])) {
            newInterval[0] = min(newInterval[0], intervals[i][0]);
            newInterval[1] = max(newInterval[1], intervals[i][1]);
            i++;
        }
        solution.push_back(newInterval);

        while (i < intervals.size()) {
            solution.push_back(intervals[i++]);
        }

        return solution;        
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.java
class Solution {
    public int maxProfit(int[] prices) {
        int lowestPriceSoFar = Integer.MAX_VALUE, maxProfit = 0;

        for (int price : prices) {
            lowestPriceSoFar = Math.min(lowestPriceSoFar, price);
            maxProfit = Math.max(maxProfit, price - lowestPriceSoFar);
        }
        return maxProfit;
    }
}
